lib/rubocop/target_finder.rb

```diff
@@ -97,7 +97,9 @@ def wanted_dir_patterns(base_dir, exclude_pattern, flags)
       base_dir = base_dir.gsub('/{}/', '/\{}/')
       dirs = Dir.glob(File.join(base_dir.gsub('/**/', '/\**/'), '*/'), flags)
                 .reject do |dir|
-                  dir.end_with?('/./', '/../') || File.fnmatch?(exclude_pattern, dir, flags)
+                  dir.end_with?('/./', '/../') ||
+                    File.fnmatch?(exclude_pattern, dir, flags) ||
+                    File.fnmatch?(exclude_pattern, "#{File.realpath(dir)}/", flags)
```

Comment:
I think line 101 is redundant here, and that might help performance. I ran your benchmark and got this:

master:
```
$ bundle exec ruby benchmark.rb
       user     system      total        real
  68.409056  14.974495  83.383551 ( 84.883239)
```

patched:
```
$ bundle exec ruby benchmark.rb
       user     system      total        real
  67.697482  15.706053  83.403535 ( 84.581582)
```

Reply:
Great point!

Reply:
That would make it impossible to exclude paths by their symlink names though, and would change the current behavior. I've added a test for this scenario in a new commit.

Reply:
Alright, thanks for that added test.

We should still avoid doing multiple `File.fnmatch?` when `dir` is not a symlink (which is probably the majority case by far). Something like (could be cleaned up probably):

```ruby
dirs = Dir.glob(File.join(base_dir.gsub('/**/', '/\**/'), '*/'), flags)
          .reject do |dir|
            next true if dir.end_with?('/./', '/../')
          
            if File.symlink?(dir.chomp('/'))
              File.fnmatch?(exclude_pattern, dir, flags) ||
                File.fnmatch?(exclude_pattern, "#{File.realpath(dir)}/", flags)
            else
              File.fnmatch?(exclude_pattern, dir, flags)
            end
          end
```

```
       user     system      total        real
  66.163276  14.268255  80.431531 ( 81.158242)
```

Reply:
The symlink check makes sense. I've updated the PR with your suggestion.